framework/validators/EachValidator.php

```diff
@@ -129,7 +129,9 @@ public function validateAttribute($model, $attribute)
         $filteredValue = [];
         foreach ($value as $k => $v) {
             $model->$attribute = $v;
-            $validator->validateAttribute($model, $attribute);
+            if (!$validator->skipOnEmpty || !$validator->isEmpty($v)) {
```

Comment:
It will be easier to read the following condition:

`!($validator->skipOnEmpty && $validator->isEmpty($v)`